.github/workflows/publish-nightly-canary.yml

```diff
@@ -49,7 +49,7 @@ jobs:
         run: git reset --hard # fix pnpm install add new line for package.json
 
       - name: Download Binding Artifacts
-        uses: actions/download-artifact@cc203385981b70ca67e1cc392babf9cc229d5806 # v4.1.9
+        uses: actions/download-artifact@95815c38cf2ff2164869cbab79da8d1f422bc89e # v4.2.1
```

Comment:
Do we need to pin the action version instead of let it update patch version? cc @Boshen